packages/app-store/zapier/api/subscriptions/addSubscription.ts

```diff
@@ -1,26 +1,15 @@
 import type { NextApiRequest, NextApiResponse } from "next";
+import { validateAccountOrApiKey } from "zapier/lib/validateAccountOrApiKey";
 
-import findValidApiKey from "@calcom/features/ee/api-keys/lib/findValidApiKey";
 import { addSubscription } from "@calcom/features/webhooks/lib/scheduleTrigger";
 import { defaultHandler, defaultResponder } from "@calcom/lib/server";
 
 async function handler(req: NextApiRequest, res: NextApiResponse) {
-  const apiKey = req.query.apiKey as string;
-
-  if (!apiKey) {
-    return res.status(401).json({ message: "No API key provided" });
-  }
-
-  const validKey = await findValidApiKey(apiKey, "zapier");
-
-  if (!validKey) {
-    return res.status(401).json({ message: "API key not valid" });
-  }
-
   const { subscriberUrl, triggerEvent } = req.body;
-
+  const { account, appApiKey } = await validateAccountOrApiKey(req, ["READ_BOOKING", "READ_PROFILE"]);
```

Comment:
Added some DRYness in here. @CarinaWolli hope you don't mind 🙏🏽

Reply:
Thank you, I also added it to deleteSubscription.ts 🙏